Add table aws_cost_by_tag closes #1532 (#1536)
turbot/steampipe-plugin-aws: docs/tables/aws_cost_by_tag.md

# Table: aws_cost_by_tag

> Amazon Cost Explorer helps you visualize, understand, and manage your AWS costs and usage. The `aws_cost_by_tag` table provides a simplified view of cost by tags in your account. ou must specify a granularity (`MONTHLY`, `DAILY`), `tag_key_1` is required, and `tag_key_2` is optional for the table.
> 

Reviewer: ```suggestion
Amazon Cost Explorer helps you visualize, understand, and manage your AWS costs and usage. The `aws_cost_by_tag` table provides a simplified view of cost by tags in your account. One must specify a granularity (`MONTHLY`, `DAILY`) and `tag_key_1` to query the table, however, `tag_key_2` is optional.
```

## Examples

### Basic info

```sql
select
  tag_key_1,
  tag_value_1,
  period_start,
  blended_cost_amount::numeric::money,
  unblended_cost_amount::numeric::money,
  amortized_cost_amount::numeric::money,
  net_unblended_cost_amount::numeric::money,
  net_amortized_cost_amount::numeric::money
from
  aws_cost_by_tag
where
  granularity = 'DAILY'
and
  tag_key_1 = 'Name';
```

### Min, Max, and average daily unblended_cost_amount by tag

```sql
select
  tag_key_1,
  tag_value_1,
  min(unblended_cost_amount)::numeric::money as min,
  max(unblended_cost_amount)::numeric::money as max,
  avg(unblended_cost_amount)::numeric::money as average
from
  aws_cost_by_tag
where
  granularity = 'DAILY'
and
  tag_key_1 = 'Name'
group by
  tag_key_1, tag_value_1;
```

### Ranked - Top 10 Most expensive days (unblended_cost_amount) by tag

```sql
with ranked_costs as (
  select
    tag_key_1,
    tag_value_1,
    period_start,
    unblended_cost_amount::numeric::money,
    rank() over(partition by tag_key_1 order by unblended_cost_amount desc)
  from
  aws_cost_by_tag
where
  granularity = 'DAILY'
and
  tag_key_1 = 'Name'
)

> select * from ranked_costs where rank <= 10;

Reviewer: @ParthaI has the query been formatted using https://www.freeformatter.com/sql-formatter.html#before-output?
